pyronear/datasets/wildfire/fire_labeler.py

```diff
@@ -109,7 +109,8 @@ def run(self):
 
         length = len(self.fire_ids[self.fire_ids == -1])
         self.fire_ids[self.fire_ids == -1] = range(current_fire_id, current_fire_id + length)
-        assert self.fire_ids[self.fire_ids == -1].sum() == 0, "Singletons escaped indexation!!"
+        if self.fire_ids[self.fire_ids == -1].sum() != 0:
+            raise AssertionError("Singletons escaped indexation!")
```

Comment:
Oh I see! But then I don't see whether this is needed in this case? If we decide to process this case in the future, we can simply remove the `AssertionError` (or put it in an `else` case), just like we would have to with `assert`? _(I've already seen several references to [this](https://docs.openstack.org/bandit/latest/plugins/b101_assert_used.html) indicating that developers should switch to `AssertionError` hence my suggestion)_

I don't know how this indexation case serious is actually. If we only need to tell the user without ever halting runtime, perhaps we could switch to warnings instead?

Also, I'm curious but what do you mean about code optimization? If I read the doc correctly, the code optimization will be different only if `__debug__` is set to `False`, and in this case, the `assert` will be removed (meaning that this case will not be processed). Perhaps it's expected, as I honestly don't know the importance of this indexation case :thinking: 

Let me know what you think, you know the use case much better than I do :)

Reply:
Using `assert` is a bad practice if one uses it in case of protecting code that can fail at runtime (in this case, we raise Exceptions)
OpenStack decided to prevent every use of assert because it seems that their contributors were using it too much _enforce interface constraints_ (wrong way, tests are here for this)

Here that's not the case, the `assert` is here to say: _at this point of the algorithm, the state of this Series must be **totally** deprived of unassigned items_ 




It warns future contributors that this is important to maintain the state of this Series as such. Because it can generate hard-to-debug errors. That's why we want the program to halt in dev mode but not in production.
`warnings` are not suitable because they can be ignored, here we want the program to halt quickly, so the contributor can directly dive into the code and fix the issue. It is particularly handy when refactoring algorithms.

Concerning optimization, we want to get rid of every stuff solely dedicated to dev mode when in production. This assertion is absolutely unneeded in production, so it can be deleted.

Reply:
Fair enough, thanks for the explanation about the expected behaviour here!
I just reverted back to `assert`